tests/utf.cpp

```diff
@@ -0,0 +1,87 @@
+#include <cassert>
+#include <cstdlib>
+#include "utils.h"
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest.h"
+
+// Correct Tests
+
+struct TestSet {
+	std::string u8;
+	std::u16string u16;
+	std::u32string u32;
+};
+
+#define CONCAT2(x, y) x ## y
+#define CONCAT(x, y) CONCAT2(x, y)
+#define STRINGIZE(x) #x
+//#define TS(X) { CONCAT(u8, STRINGIZE(X)), CONCAT(u, STRINGIZE(X)), CONCAT(U, STRINGIZE(X)) }
+#define TS(X) { CONCAT(u8, X), CONCAT(u, X), CONCAT(U, X) }
+
+TestSet tests[] = {
+	//Valid Strings
+	TS("κόσμε"),
+	//First Char tests
+	TS("\U00000000"),
+	TS("\U00000080"),
+	TS("\U00000800"),
+	TS("\U00010000"),
+//	TS("\U00200000"),
+//	TS("\U04000000"),
```

Comment:
What are these commented values? Invalid codepoints?

Reply:
They are codepoints > 4 bytes. The compiler doesn't like them so they are commented out.